File: frontend/src/main/frontend/src/app/component/report/report.ts
import {AppUser} from "../users/user";

export class Report {

  id: number;
  description: string;
  filePath: string;
  fileName: string;
  fileExtension: string;
  grade: string;
  createdDate: string;
  lastModifiedDate: string;
  createdBy: string;
  lastModifiedBy: string;
  isSendInTime: boolean;
  users : AppUser[];

  constructor(json? : any) {
    if (json != null) {
      this.id = json.id;
      this.description = json.description;
      this.filePath = json.filePath;
      this.fileName = json.fileName;
      this.fileExtension = json.fileExtension;
      this.grade = json.grade;
      this.createdDate = json.createdDate;
      this.lastModifiedDate = json.lastModifiedDate;
      this.createdBy = json.createdBy;
      this.lastModifiedBy = json.lastModifiedBy;
      this.isSendInTime = json.isSendInTime;
      if (json.users != null) {
        this.users = AppUser.toArray(json.roles);
      }
    }
  }

  // Utils
  static toArray(jsons : any[]) : Report[] {
    let reports : Report[] = [];
    if (jsons != null) {
      for (let json of jsons) {
        reports.push(new Report(json));
      }
    }
    return reports;
  }
}

File: backend/src/main/java/reports/reports/domain/Report_.java
package reports.reports.domain;

import javax.persistence.metamodel.ListAttribute;
import javax.persistence.metamodel.SingularAttribute;
import javax.persistence.metamodel.StaticMetamodel;

@StaticMetamodel(Report.class)
public abstract class Report_ {

    public static volatile SingularAttribute<Report, Integer> id;
    public static volatile SingularAttribute<Report, String> description;
    public static volatile SingularAttribute<Report, String> filePath;
    public static volatile SingularAttribute<Report, String> fileName;
    public static volatile SingularAttribute<Report, String> fileExtension;
    public static volatile SingularAttribute<Report, String> grade;
    public static volatile SingularAttribute<Report, Long> createdDate;
    public static volatile SingularAttribute<Report, Long> lastModifiedDate;
    public static volatile SingularAttribute<Report, String> createdBy;
    public static volatile SingularAttribute<Report, String> lastModifiedBy;
    public static volatile SingularAttribute<Report, Boolean> isSendInTime;

    public static volatile ListAttribute<Report, AppUser> users;

    public static volatile SingularAttribute<Report, Laboratory> laboratory;
}
